marconi-tutorial/src/Marconi/Tutorial/HttpServer.hs

```diff
@@ -35,6 +37,62 @@ type RpcMethod = RpcAddressCountMethod
 type RpcAddressCountMethod =
   JsonRpc
     "getAddressCount"
+    AddressCountQuery
     String
-    String
-    Int
+    (Core.Result AddressCountQuery)
+
+{----------------------
+   API implementation
+-----------------------}
+
+runHttpServer :: ReaderT Env IO ()
+runHttpServer = do
+  env <- ask
+  let cliOptions = view envCliArgs env
+  let httpSettings = maybe id setPort (optionsHttpPort cliOptions) defaultSettings
+  liftIO $ runSettings httpSettings (httpApp env)
+
+httpApp
+  :: Env
+  -> Application
+httpApp env = serve (Proxy @API) (httpServer env)
+
+httpServer
+  :: Env
+  -> Server API
+httpServer = getAddressCountHttpHandler
+
+getAddressCountHttpHandler
+  :: Env
+  -> AddressCountQuery
+  -> Handler (Either (JsonRpcErr String) (Core.Result AddressCountQuery))
+getAddressCountHttpHandler env =
+  queryIndexerVarHandler (Core.standardWorkerIndexerVar $ env ^. addressCountIndexerWorker)
+
+queryIndexerVarHandler
+  :: ( Show (Core.Result query)
+     , Ord (Core.Point event)
+     , Core.Queryable (ExceptT (Core.QueryError query) Handler) event query indexer
+     , Core.IsSync (ExceptT (Core.QueryError query) Handler) event indexer
+     )
+  => MVar (indexer event)
+  -> query
+  -> Handler (Either (JsonRpcErr String) (Core.Result query))
+queryIndexerVarHandler indexerVar q = do
+  indexer <- liftIO $ readMVar indexerVar
+  toHttpHandler indexer q
+
+toHttpHandler
+  :: ( Show (Core.Result query)
+     , Ord (Core.Point event)
+     , Core.Queryable (ExceptT (Core.QueryError query) Handler) event query indexer
+     , Core.IsSync (ExceptT (Core.QueryError query) Handler) event indexer
+     )
+  => indexer event
+  -> query
+  -> Handler (Either (JsonRpcErr String) (Core.Result query))
+toHttpHandler indexer q = do
+  countE <- runExceptT $ Core.queryLatest q indexer
+  case countE of
+    Left err -> pure $ Left $ mkJsonRpcInternalErr (Just $ show err)
+    Right count -> pure $ Right count
```

Comment:
Neat!

Reply:
FYI, you could use `first` here:
```
toHttpHandler indexer q =
  first (mkJsonRpcInternalErr . Just . show) <$>
    runExceptT (Core.queryLatest q indexer)
```
but maybe that's too terse.

Reply:
Will we move it to the future `marconi-json-rpc` library? It looks like a highly reusable piece of code now.

Reply:
> FYI, you could use first here:

True, but readability takes a hit.

> Will we move it to the future marconi-json-rpc library?

Yep! I'll make another PR for that.